### PR TITLE
[Datastore] Fix Stream Storey wrapper

### PR DESCRIPTION
**Pull request by alxtkr77**
https://iguazio.atlassian.net/browse/ML-7904

### CODE REVIEW

File: mlrun/datastore/storeytargets.py [hunk 1]
```diff
@@ -83,11 +84,13 @@ def __init__(self, *args, **kwargs):
         args = list(args)
 
         path = args[0] if args else kwargs.get("stream_path")
-        endpoint, storage_options = get_url_and_storage_options(path)
 
         if not path:
             raise mlrun.errors.MLRunInvalidArgumentError("StreamTarget requires a path")
 
+        _, storage_options = get_url_and_storage_options(path)
+        endpoint, uri = parse_path(path)
```
Comment (gtopper) [inline mlrun/datastore/storeytargets.py]: It's not a URI though, it's a path. The original `path` is a URI.

File: mlrun/datastore/storeytargets.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 import mlrun.model_monitoring.helpers
 from mlrun.datastore.base import DataStore
 
+from ..platforms.iguazio import parse_path
```
Comment (gtopper) [inline mlrun/datastore/storeytargets.py:22]: Not sure you're supposed to import enterprise code into general mlrun code.

**Review by gtopper — COMMENTED** [review]
Any chance we can have a test for this?

File: mlrun/datastore/storeytargets.py (hunk 2 above)
Comment (alxtkr77) [reply mlrun/datastore/storeytargets.py:22]: targets.py does it

Comment (gtopper) [reply mlrun/datastore/storeytargets.py:22]: Oh well then.

File: tests/system/feature_store/test_feature_store.py [hunk 3]
```diff
@@ -5048,6 +5049,25 @@ def test_snowflake_target_error(self, local):
...
+                    datetime(2024, 9, 19, 16, 22, 9, 53003),
+                ],
+                "clunfoyd": [0.339612325, 0.3446700093, 0.9394242442],
+            }
+        )
+
+        target = StreamTarget(
+            path=f"v3io:///projects/{self.project_name}/test_stream_target",
+            name="lctcnnzgrg",
+            attributes={"shards": 1},
```
Comment (gtopper) [inline tests/system/feature_store/test_feature_store.py]:
```suggestion
```

File: tests/system/feature_store/test_feature_store.py [hunk 4]
```diff
@@ -5048,6 +5049,25 @@ def test_snowflake_target_error(self, local):
...
 
+    def test_stream_target(self):
+        source = pd.DataFrame(
+            {
+                "time_stamp": [
+                    datetime(2024, 9, 19, 16, 22, 7, 51001),
+                    datetime(2024, 9, 19, 16, 22, 8, 52002),
+                    datetime(2024, 9, 19, 16, 22, 9, 53003),
+                ],
+                "clunfoyd": [0.339612325, 0.3446700093, 0.9394242442],
```
Comment (gtopper) [inline tests/system/feature_store/test_feature_store.py]:
```suggestion
                "key": [0.339612325, 0.3446700093, 0.9394242442],
```

**Review by gtopper — COMMENTED** [review]
LGTM other than these two minor comments.

**Comment by gtopper** [comment]
Also, I guess target branch needs to be changed to `1.7.x`, if I understand the current workflow correctly.

**Review by assaf758 — APPROVED** [review]

**Pull request merged**